internal/mode/static/state/graph/graph.go

```diff
@@ -78,6 +83,17 @@ func (g *Graph) IsReferenced(resourceType client.Object, nsname types.Namespaced
 		_, existed := g.ReferencedNamespaces[nsname]
 		exists := isNamespaceReferenced(obj, g.Gateway)
 		return existed || exists
+	// Service reference exists if at least one HTTPRoute references it.
+	case *v1.Service:
```

Comment:
A couple cases that I've thought of:

1. HTTPRoute exists and references an existing Service - Service will be in ReferencedServicesNames and since Services cannot be "renamed" any other changes to the Service will be caught here.
2. HTTPRoute exists and a Service is created that matches the HTTPRoute - ReferencedServicesNames will already have the Service name thus it will trigger a change.
3. HTTPRoute does not exist and a Service is created - ReferencedServicesNames will not have the Service name thus it will not trigger here.
4. HTTPRoute exists and references an existing Service and the HTTPRoute changes to match/unmatch - Since the HTTPRoute changed, it will trigger a graph update and nginx configuration update itself.

Does this logic seem work?